tools/default_software.py

```diff
@@ -0,0 +1,43 @@
+__all__ = ["get_returnn_root"]
+
+import logging
+from typing import Optional
+
+from sisyphus import gs, Path
+
+import i6_core.tools as tools
+
+RETURNN_COMMIT = None
+
+
+def get_returnn_root(returnn_root: Optional[Path]) -> Path:
+    """
+    Provides a tk.Path object that points to the root of a returnn repository.
+    Will return the first location it finds:
+        1. the argument provided to get_returnn_root
+        2. gs.RETURNN_ROOT
+        3. checkout current head
+
+    :param returnn_root: Path to the root of a returnn repository or None
+    :returns: Path to the root of a returnn repository
+    """
+    if isinstance(returnn_root, Path):
+        return returnn_root
+    elif isinstance(returnn_root, str):
+        logging.warning("Giving returnn_root as str is deprecated, use tk.Path instead")
+        return Path(returnn_root)
+    elif returnn_root is None:
+        if hasattr(gs, "RETURNN_ROOT"):
+            logging.warning(
+                "Using gs.RETURNN_ROOT is deprecated, please provide an explicit tk.Path object"
+            )
+            return Path(gs.RETURNN_ROOT)
+        else:
+            returnn_root = tools.CloneGitRepositoryJob(
```

Comment:
At first I thought we should add a warning also in this branch, that the current HEAD at the time of first running is used which leads to inconsistencies. But then maybe 
  1. this warning should be added to `CloneGitRepositoryJob` in general
  2. we would like to provide a means to set a "default commit" for the default returnn

So therefor we now have a `i6_core.tools.default_software.RETURNN_COMMIT` constant. A user could now in principle do 
```
import i6_core.tools.default_software as ds
ds.RETURNN_COMMIT = "abcdef"
```
this might even work well in 99% of the cases, but if anybody ever does
```
sis c config/here_I_set_RETURNN_COMMIT.py config/here_I_dont.py
tk.cleaner.clean_work_dir(mode="remove")
```
then all trainings in the second config will be deleted. So maybe we should not facilitate this error by letting the user provide a different commit than HEAD?